### PR TITLE
Closes #25 Add a draft of ADAPET and ADTPET template

### LINKED ISSUE #25: Showcase `compute_centiloid()` in ADAPET template
### Background Information

Now that the draft ADPET template is merged, we can showcase `compute_centiloid()` within the amyloid variant of the template.

### Definition of Done

- [ ] ADPET template split out into amyloid (ADAPET) and tau (ADTPET) templates, with headers and language updated where necessary.
- [ ] ADAPET template enhanced with example application of `admiralneuro::compute_centiloid()` to derive new centiloid records. May need to use in combination with `derive_param_computed()`.

### PR DESCRIPTION
**Pull request by meilinjiang**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @meilinjiang!! I've left some initial comments. Additionally, I wonder iof you can take the opportunity to please rename the other template to `adtpet` as well? And implement tau filtering only? Many thanks, and please let me know of any questions.

File: inst/templates/ad_adapet.R [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+          NVORRES = as.character(AVAL),
+          NVORRESU = AVALU,
+          NVSTRESC = as.character(AVAL),
+          NVSTRESN = AVAL,
+          NVSTRESU = AVALU
+        )
+      )
+    ),
+    derivation_slice(
+      filter = (PARAMCD == "SUVRFBP") & (NVMETHOD == "BERKELEY FBP SUVR PIPELINE") & (REFREG == "Whole Cerebellum"),
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:137]: Yes I saw this warning message as well. Our example data doesn't have this particular filter condition but it is one of the acceptable combination of compute_centiloid() so I included it here. Any other thoughts are welcome.

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_adapet.R:137]: that makes sense, in which case please leave commented out with some sort of text explaining that if the user has this combination, they can use this code. We can't have any warnings in our templates so we can't leave as is

[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:137]: Agree. Will fix it

File: inst/templates/ad_adapet.R [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    AVALC = if_else(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
+      NVSTRESC,
+      NA
+    )
+  )
+
+## Convert SUVR to Centiloid
+adapet <- adpet %>%
+  filter(AGCAT == "AMYLOID TRACER") %>%
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Agree. Fixed

File: inst/templates/ad_adapet.R [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+    args = params(
+      by_vars = exprs(
+        STUDYID, DOMAIN, USUBJID, NVSEQ, NVLNKID, NVLOC, NVMETHOD, NVNAM, NVCAT,
+        VISITNUM, VISIT, NVDTC, NVDY, REFREG, IDVARVAL,
+        NVLOBXFL, TRTSDT, TRTEDT, TRT01A, TRT01P,
+        AGTRT, AGCAT, ADT, ADY
+      ),
+      constant_by_vars = NULL,
+      constant_parameters = NULL,
+      keep_nas = T
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Fixed

File: inst/templates/ad_adapet.R [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+  slice_derivation(
+    derivation = derive_param_computed,
+    args = params(
+      by_vars = exprs(
+        STUDYID, DOMAIN, USUBJID, NVSEQ, NVLNKID, NVLOC, NVMETHOD, NVNAM, NVCAT,
+        VISITNUM, VISIT, NVDTC, NVDY, REFREG, IDVARVAL,
+        NVLOBXFL, TRTSDT, TRTEDT, TRT01A, TRT01P,
+        AGTRT, AGCAT, ADT, ADY
+      ),
+      constant_by_vars = NULL,
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Fixed

File: inst/templates/ad_adapet.R [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+          NVTESTCD = "CLFBB_BERKELEY",
+          NVTEST = "Centiloid (CL) based on BERKELEY FBB",
+          PARAMCD = "CLFBB_BERKELEY",
+          PARAM = "Centiloid (CL) based on BERKELEY FBB",
+          AVALU = "CL",
+          NVORRES = as.character(AVAL),
+          NVORRESU = AVALU,
+          NVSTRESC = as.character(AVAL),
+          NVSTRESN = AVAL,
+          NVSTRESU = AVALU
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Fixed

File: inst/templates/ad_adapet.R [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+        set_values_to = exprs(
+          AVAL = compute_centiloid(
+            tracer = "18F-Florbetaben",
+            pipeline = "BERKELEY FBB SUVR PIPELINE",
+            ref_region = "Whole Cerebellum",
+            suvr = AVAL
+          ),
+          NVTESTCD = "CLFBB_BERKELEY",
+          NVTEST = "Centiloid (CL) based on BERKELEY FBB",
+          PARAMCD = "CLFBB_BERKELEY",
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Fixed

File: inst/templates/ad_adapet.R [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+          NVSTRESC = as.character(AVAL),
+          NVSTRESN = AVAL,
+          NVSTRESU = AVALU
+        )
+      )
+    )
+  )
+
+
+## Derive amyloid positivity based on cutoff of 24.1 CL
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Removed

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
@meilinjiang thanks for the updates! Looking much better now. Just left a few more comments, please also remember to delete ad_adpet.R

**Comment by manciniedoardo** [comment]
@meilinjiang also please make sure to document the datasets you are storing in `data.R` - this is why the checks are failing

File: inst/templates/ad_adapet.R [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+            tracer = "18F-Florbetaben",
+            pipeline = "BERKELEY FBB SUVR PIPELINE",
+            ref_region = "Whole Cerebellum",
+            suvr = AVAL
+          ),
+          PARAMCD = "CLBFBB",
+          PARAM = "Centiloid (CL) based on BERKELEY FBB",
+          AVALU = "CL",
+          AVISIT = VISIT,
+          AVISITNUM = VISITNUM,
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:123]: Thanks of reviewing again! It is helpful. 
For this one, we already have AVISITN in the dataset as the week number so I used AVISITNUM instead. We could further discuss it if this variable is necessary in our meeting today.

File: inst/templates/ad_adapet.R [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+  )
+
+## Convert SUVR to Centiloid
+adapet <- adapet %>%
+  slice_derivation(
+    derivation = derive_param_computed,
+    args = params(
+      by_vars = c(
+        get_admiral_option("subject_keys"),
+        exprs(TRT01A, TRT01P, ADT, ADY, TRTSDT, TRTEDT)
```
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:104]: Are you referring to "ADT" and "ADY" variables? Without populating the "ADT", we could not get the Baseline, Change etc for derived amyloid records

File: inst/templates/ad_adapet.R [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+    new_vars = exprs(AGTRT, AGCAT),
+    by_vars = exprs(USUBJID, VISIT, NVLNKID = AGLNKID)
+  ) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = NVDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  filter(AGCAT == "AMYLOID TRACER")
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply inst/templates/ad_adapet.R]: then i would add something like:
```
ag_amyloid <- ag %>% 
  filter(AGCAT == "AMYLOOID TRACER")
```
near line 37

File: inst/templates/ad_adtpet.R [hunk 11]
```diff
@@ -54,28 +54,29 @@ attr(param_lookup$NVTESTCD, "label") <- "NV Test Short Name"
...
   ) %>%
-  ## Join ADPET with AG for tracer information ----
+  ## Join ADTPET with AG for tracer information ----
   # Users can add more variables in the `new_vars` argument as needed.
   derive_vars_merged(
     dataset_add = ag,
     new_vars = exprs(AGTRT, AGCAT),
-    by_vars = exprs(USUBJID, VISIT, NVLNKID = AGLNKID)
+    by_vars = c(get_admiral_option("subject_keys"), exprs(VISIT, NVLNKID = AGLNKID))
   ) %>%
```
Comment (jwang-lilly) [inline inst/templates/ad_adtpet.R:70]: I think it is better to add the filter immediately after AG is merged so we don't need to add additional columns such as ADT and ADY to uneeded rows, 
filter(AGCAT == "TAU TRACER") # Filter nv dataset for tau records only

File: inst/templates/ad_adapet.R [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+# Combine the parental datasets with their respective supp datasets (only if exist)
+# User can use `combine_supp()` from {metatools} to combine the parental with supp dataset.
+nv <- metatools::combine_supp(nv, suppnv)
+
+# Lookup tables ----
+
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~NVTESTCD, ~NVCAT, ~NVLOC, ~REFREG, ~PARAMCD, ~PARAM, ~PARAMN,
+  "SUVR", "FBP", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "SUVRFBP", "FBP Standard Uptake Ratio Neocortical Composite Whole Cerebellum", 1,
```
Comment (jwang-lilly) [inline inst/templates/ad_adapet.R:44]: I see you included a letter for pipeline for the added centiloid rows.  Maybe you can do the same here:
SUVRFBP => SUVRAFBP
SUVRFBB => SUVRBFBP
SUVRFTP => SUVRBFTP

File: inst/templates/ad_adapet.R [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+    by_vars = get_admiral_option("subject_keys")
+  ) %>%
+  ## Join ADAPET with AG for tracer information ----
+  # Users can add more variables in the `new_vars` argument as needed.
+  derive_vars_merged(
+    dataset_add = ag,
+    new_vars = exprs(AGTRT, AGCAT),
+    by_vars = c(get_admiral_option("subject_keys"), exprs(VISIT, NVLNKID = AGLNKID))
+  ) %>%
+  ## Calculate ADT, ADY ----
```
Comment (jwang-lilly) [inline inst/templates/ad_adapet.R:71]: move up the filter(AGCAT == "AMYLOID TRACER") here.

File: inst/templates/ad_adapet.R [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+adapet <- adapet %>%
+  derive_var_ontrtfl(
+    start_date = ADT,
+    ref_start_date = TRTSDT,
+    ref_end_date = TRTEDT,
+    filter_pre_timepoint = toupper(AVISIT) == "BASELINE" # Observations as not on-treatment
+  )
+
+
+### Derive Baseline flags ----
```
Comment (jwang-lilly) [inline inst/templates/ad_adapet.R:193]: ###  xxxx   ---- and ##  xxxx ---- are used. ### is lower level than ##   
Is this intentional?
If yes, it makes sense.
should line 243 ### Calculate BASEC ---- ?

File: inst/templates/ad_adapet.R [hunk 15]
```diff
@@ -0,0 +1,287 @@
...
+    by_vars = get_admiral_option("subject_keys"),
+    order = exprs(PARAMCD, ADT, AVISITN, VISITNUM),
+    check_type = "error"
+  )
+
+
+# Final Steps, Select final variables and Add labels ----
+# This process will be based on your metadata, no example given for this reason
+# ...
+
```
Comment (jwang-lilly) [inline inst/templates/ad_adapet.R:275]: remove extract blank line

File: inst/templates/ad_adapet.R [hunk 16]
```diff
@@ -0,0 +1,287 @@
...
+suppnv <- admiralneuro::suppnv_neuro
+adsl <- admiralneuro::adsl_neuro
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+nv <- convert_blanks_to_na(nv)
+
+
```
Comment (jwang-lilly) [inline inst/templates/ad_adapet.R:34]: remove extra blank line

**Comment by jwang-lilly** [comment]
@meilinjiang, great work with the adapet and adtpet. I left a few minor comments. I think you haven't run `devtools::document()` to created corresponding man pages .Rd

File: inst/templates/ad_adapet.R (hunk 10 above)
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: This would only filter AG, the "TAU TRACER" in NV would still be there but with missing AGCAT. I am trying to think if there is any other way we can move up the filter before merging AG with NV

File: inst/templates/ad_adapet.R (hunk 12 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:44]: Yes I included a letter for pipeline for the added centiloid rows to differentiate multiple centiloid records per patient at the same visit. In our example dataset, a patient may have more than one SUVR pipeline per visit. 
I have added the a letter to represent pipeline for SUVR as well.

File: inst/templates/ad_adapet.R (hunk 14 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:193]: @manciniedoardo We may need your help here. Is there a guidance on the level of usage for ## and ###?

File: inst/templates/ad_adapet.R (hunk 15 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:275]: removed line275

File: inst/templates/ad_adapet.R (hunk 16 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:34]: removed line 33

File: inst/templates/ad_adapet.R [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+  mutate(
+    AVAL = NVSTRESN,
+    AVALC = if_else(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
+      NVSTRESC,
+      NA
+    )
+  )
+
+## Convert SUVR to Centiloid
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:96]:
```suggestion
## Convert SUVR to Centiloid ----
```

File: inst/templates/ad_adapet.R [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+      by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD, ABLFL)),
+      order = exprs(ADT),
+      mode = "last"
+    ),
+    filter = !is.na(AVISITN) & (ONTRTFL == "Y" | ABLFL == "Y")
+  )
+
+## Derive baseline information ----
+
+## Calculate BASE ----
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:233]:
```suggestion
### Calculate BASE ----
```

File: inst/templates/ad_adapet.R [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+adapet <- adapet %>%
+  derive_var_base(
+    by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD, BASETYPE)),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  ## Calculate BASEC ----
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:240]:
```suggestion
  ### Calculate BASEC ----
```

File: inst/templates/ad_adapet.R [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+    source_var = AVALC,
+    new_var = BASEC
+  ) %>%
+  ## Calculate CHG for post-baseline records ----
+  # The decision on how to populate pre-baseline and baseline values of CHG is left as a user choice
+  restrict_derivation(
+    derivation = derive_var_chg,
+    filter = AVISITN > 0
+  ) %>%
+  ## Calculate PCHG for post-baseline records ----
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:252]:
```suggestion
  ### Calculate PCHG for post-baseline records ----
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Added suggested sectioning to remain consistent - @meilinjiang please mimic in ADTPET template!

File: inst/templates/ad_adapet.R (hunk 14 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R:190]: Agree to be more consistent. Fixed

File: inst/templates/ad_adapet.R (hunk 18 above)
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: Fixed in both templates

**Comment by meilinjiang** [comment]
> @meilinjiang can you try run `devtools::document()` to fix the failing checks?

File: inst/templates/ad_adapet.R (hunk 10 above)
[the remark this answers is not in the document]
Comment (meilinjiang) [reply inst/templates/ad_adapet.R]: I was thinking we can filter "NVMETHOD" in NV and "AGCAT" in AG if we want to filter earlier. The only concern is that "NVMETHOD" might have more variations than "TAU TRACER" category in AG

**Review by manciniedoardo — APPROVED** [review]
@meilinjiang looks good to me - @jwang-lilly please takke a final look, then we can merge.

**Comment by jwang-lilly** [comment]
@meilinjiang, @manciniedoardo, look great!

**Comment by cgpele** [comment]
Hi @jwang-lilly @meilinjiang @manciniedoardo 

I just have a question regarding the PARAM/PARAMCD for the centiloids (and sorry, I may have missed this discussion in recent weeks), hence not re-opening this - just adding a comment for discussion

We have differen PARAMCD/PARAM for Centiloids derived from different pipelines, but isn't the point here that centiloids allows for analysis across tracers and pipelines.  Wouldn't we want the same PARAMCD/PARAM then?

image

**Comment by jwang-lilly** [comment]
@cgpele, I just saw this comment. You’re absolutely right! Using a single PARAMCD/PARAM would make the ADaM more “analysis-ready,” so we wouldn’t need to filter with PARAMCD %in% c(set1, set2, …). 

Would you be willing to raise this as an issue for Phase 2 improvements, or at least bring it up at the team meeting? Lilly has adopted the approach you suggested, and I can see the advantages.

In my work (examining Centiloid and visual interpretation of scans) I need to distinguish FBP from FBB scans as they are visually different. Tracking which subjects had FBB or FBP at which visits adds quite a few extra steps. But for mainstream analyses, having a single, unified Centiloid parameter would be ideal.

**Comment by manciniedoardo** [comment]
@jwang-lilly I believe that @cgpele is on leave so i suggest making the issue yourself so that this doesn't get lost! looks like a great improvement
